feature/gnoi/packet_link_qualification/tests/packet_link_qualification_test/packet_link_qualification_test.go

```diff
@@ -337,23 +362,70 @@ func testLinkQualification(t *testing.T, dut1 *ondatra.DUTDevice, dut2 *ondatra.
+		// reflectorSetupDuration = capabilities.GetReflector().GetPmdLoopback().GetMinSetupDuration().AsDuration()
+		// reflectorTeardownDuration = capabilities.GetReflector().GetPmdLoopback().GetMinTeardownDuration().AsDuration()
+	} else if asicLB := ref.GetAsicLoopback(); asicLB.GetMinSetupDuration().GetSeconds() >= 1 && asicLB.GetMinTeardownDuration().GetSeconds() >= 1 {
+		t.Logf("Device supports ASIC loopback reflector mode")
+		// reflectorSetupDuration = capabilities.GetReflector().GetAsicLoopback().GetMinSetupDuration().AsDuration()
+		// reflectorTeardownDuration = capabilities.GetReflector().GetAsicLoopback().GetMinTeardownDuration().AsDuration()
+	} else {
+		t.Errorf("Reflector MinSetupDuration or MinTeardownDuration is not >=1 for supported mode. Device reflector capabilities")
+	}
+
+	// generatorSetupDuration := capabilities.GetGenerator().GetPacketGenerator().GetMinSetupDuration().AsDuration()
+	// generatorTeardownDuration := capabilities.GetGenerator().GetPacketGenerator().GetMinTeardownDuration().AsDuration()
+
+	generatorPlqResp, err := gnoiClient.LinkQualification().Capabilities(context.Background(), &plqpb.CapabilitiesRequest{})
+	t.Logf("LinkQualification().Capabilities(): %v, err: %v", generatorPlqResp, err)
+	if err != nil {
+		t.Fatalf("Failed to handle gnoi LinkQualification().Capabilities(): %v", err)
 	}
+
+	reflectorPlqResp, err := gnoiClient.LinkQualification().Capabilities(context.Background(), &plqpb.CapabilitiesRequest{})
+	t.Logf("LinkQualification().Capabilities(): %v, err: %v", reflectorPlqResp, err)
+	if err != nil {
+		t.Fatalf("Failed to handle gnoi LinkQualification().Capabilities(): %v", err)
+	}
+
+	genPblqMinSetup := float64(generatorPlqResp.GetGenerator().GetPacketGenerator().GetMinSetupDuration().GetSeconds())
+	refPblqMinSetup := float64(reflectorPlqResp.GetGenerator().GetPacketGenerator().GetMinSetupDuration().GetSeconds())
+	genPblqMinTearDown := float64(generatorPlqResp.GetGenerator().GetPacketGenerator().GetMinTeardownDuration().GetSeconds())
+	refPblqMinTearDown := float64(reflectorPlqResp.GetGenerator().GetPacketGenerator().GetMinTeardownDuration().GetSeconds())
+
 	plqDuration := &LinkQualificationDuration{
 		generatorpreSyncDuration:  30 * time.Second,
 		reflectorpreSyncDuration:  0 * time.Second,
-		generatorsetupDuration:    30 * time.Second,
-		reflectorsetupDuration:    60 * time.Second,
+		generatorsetupDuration:    time.Duration(math.Max(30, genPblqMinSetup)) * time.Second,
+		reflectorsetupDuration:    time.Duration(math.Max(60, refPblqMinSetup)) * time.Second,
 		testDuration:              120 * time.Second,
 		generatorPostSyncDuration: 5 * time.Second,
 		reflectorPostSyncDuration: 10 * time.Second,
-		tearDownDuration:          30 * time.Second,
+		generatorTeardownDuration: time.Duration(math.Max(30, genPblqMinTearDown)) * time.Second,
+		reflectorTeardownDuration: time.Duration(math.Max(30, refPblqMinTearDown)) * time.Second,
```

Comment:
Thanks for doing this, the logic looks good.  Please remove the comments and move this code into a function call (such as `func setPlqTimers` so the code is easier to read.